src/game/board.rs

```diff
@@ -82,7 +82,7 @@ impl Board {
                 // lock piece and push any full lines to vec_full_lines
                 self.vec_active_piece[player as usize].shape = Shapes::None;
                 let mut is_full_line = false;
-                for row in self.lock_piece(player).iter() {
+                for row in &self.lock_piece(player) {
```

Comment:
is this something that just sorta works better for vectors?

Reply:
Usually you use the &, unless you are using other iterator functions